frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -0,0 +1,61 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.DocumentAdapter
+import com.intellij.ui.layout.CellBuilder
+import com.intellij.ui.layout.ValidationInfoBuilder
+import com.intellij.ui.layout.panel
+import com.virtuslab.gitmachete.frontend.actions.common.GitMacheteBundle.message
+import kotlin.apply
+import kotlin.text.isEmpty
+import kotlin.text.trim
+
+class SlideInDialog constructor(project: Project,
+                                private var parentName: String)
+  : DialogWrapper(project, true) {
+
+  private var branchName = ""
```

Comment:
Nit: drop a comment that this field is only ever meant to be written on UI thread (IIUC), so no synchronization is needed